request.js

```diff
@@ -74,14 +74,14 @@
             request.onload = function(){
                 if((request.statusText === 'OK' && request.status === 200) || typeof request.statusText === 'undefined'){
                     methods.success.apply(methods, utils.parse(request));
-                    methods.always.apply();    
+                    methods.always.apply();
                 } else {
                     methods.error.apply(methods, utils.parse(request));
                     methods.always.apply();
                 }
             };
-            request.onerror = function(){
-                methods.error.apply(methods, utils.parse(request));
+            request.onerror = function(e){
+                methods.error.apply(methods, e);
```

Comment:
I can change it if you want but I believe is irrelevant. According to the *XMLHttpRequest* API documentation, *onerror* is triggered by network issues (like a failure on SSL handshake, a connections refused, a DNS resolution error, etc) or a denied request (like a failure on meeting a CORS requirement). In any case, you wouldn't have a response. If an error occurs on application level (like HTTP, e.g, an status code equal to 500) you would have a response and *onload* would be triggered.